### DISCUSSION TITLE
Render future predictions

### DISCUSSION THREAD

**Discussion by amacneil** [foxglove/community#302] [Q&A]
In some cases it is desired to render predictions, e.g. the future robot pose or predicted markers at the current timestamp.

It is possible to achieve this today using the Variable Slider panel with a Node Playground script to publish markers on a new topic. However, this is clunky. It would be great to have a concept of rendering predictions natively in Studio.

Streetscape.gl appears to have a "look ahead" feature integrated with their scrub bar:

<img width="687" alt="image" src="https://user-images.githubusercontent.com/637671/168870135-2c9fc882-9a02-4b7a-9335-9ec051e0d6be.png">

**Comment by defunctzombie** [comment]
Might be worth describing this feature more specifically around having a control for such rendering. You can already render "future" predictions by drawing markers (of various opacity and color) for your predictions. So that ask here seems to be more around dynamically adjusting the amount of markers rendered.

**Comment by amacneil** [comment]
Example of future time with NP + variables in Webviz here:

https://www.youtube.com/watch?v=7myZAhC4Uuw&t=195s
